internal/dialectadapter/locker.go

```diff
@@ -1,139 +0,0 @@
-package dialectadapter
-
-import (
-	"context"
-	"database/sql"
-	"errors"
-
-	"github.com/pressly/goose/v4/internal/dialectadapter/dialectquery"
-	"github.com/sethvargo/go-retry"
-)
-
-var (
-	// defaultLockID is the id used to lock the database for migrations. It is a crc64 hash of the
-	// string "goose". This is used to ensure that the lock is unique to goose.
-	//
-	// crc64.Checksum([]byte("goose"), crc64.MakeTable(crc64.ECMA))
-	defaultLockID int64 = 5887940537704921958
-
-	// ErrLockNotImplemented is returned when the database does not support locking.
-	ErrLockNotImplemented = errors.New("lock not implemented")
-)
-
-// Locker defines the methods to lock and unlock the database.
-//
-// Locking is an experimental feature and the underlying implementation may change in the future.
-//
-// The only database that currently supports locking is Postgres. Other databases will return
-// ErrLockNotImplemented.
-type Locker interface {
-	// CanLock returns true if the database supports locking.
-	CanLock() bool
-
-	// LockSession and UnlockSession are used to lock the database for the duration of a session.
-	//
-	// The session is defined as the duration of a single connection and both methods must be called
-	// on the same connection.
-	LockSession(ctx context.Context, conn *sql.Conn) error
-	UnlockSession(ctx context.Context, conn *sql.Conn) error
-
-	// LockTransaction is used to lock the database for the duration of a transaction.
-	// LockTransaction(ctx context.Context, tx *sql.Tx) error
-}
-
-func (s *store) CanLock() bool {
-	switch s.querier.(type) {
-	case *dialectquery.Postgres:
-		return true
-	default:
-		return false
-	}
-}
-
-func (s *store) LockSession(ctx context.Context, conn *sql.Conn) error {
-	var fn func(context.Context) error
-
-	switch t := s.querier.(type) {
-	case *dialectquery.Postgres:
```

Comment:
IMO it would not scale well - it will be a lot of switch cases.

Instead, we can just provide multiple interface implementations later.